### PR TITLE
Do not update a scaffold's AppBar scroll offset if the drawer is open

### LINKED ISSUE #5131: Gallery, Landscape: Scrolling drawer also scrolls background
On iOS, open Gallery, turn to landscape, and then scroll inside of drawer.

Expected: background doesn't move.

Actual: background moves.

### PR DESCRIPTION
**Pull request by jason-simmons**
Fixes https://github.com/flutter/flutter/issues/5131

### CODE REVIEW

**Comment by dragostis** [comment]
LGTM.

File: packages/flutter/lib/src/material/scaffold.dart [hunk 1]
```diff
@@ -591,6 +591,10 @@ class ScaffoldState extends State<Scaffold> {
 
   bool _handleScrollNotification(ScrollNotification notification) {
     final ScrollableState scrollable = notification.scrollable;
+
+    if (_drawerKey.currentState?.isOpen)
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/scaffold.dart]: if (_drawerKey.currentState == null ||  _drawerKey.currentState.isOpen)

**Comment by HansMuller** [comment]
I'm not sure that this the right fix. The source of the problem is the lack of a scrollableKey for the Scaffold. Without that key, any scrollable that ends up on top of the appbar will generate scroll notifications that propel the appbar's animation.

We could avoid this problem and some similar ones by putting the scroll notification listener around the scaffold's body, instead of being effectively around the entire scaffold.

File: packages/flutter/lib/src/material/scaffold.dart (hunk 1 above)
Comment (dragostis) [inline packages/flutter/lib/src/material/scaffold.dart]: `_drawerKey.currentState?.isOpen ?? true` should also work.

**Comment by jason-simmons** [comment]
Moved the scroll notification listener to the body widget.

PTAL

File: packages/flutter/lib/src/material/scaffold.dart [hunk 2]
```diff
@@ -728,27 +740,15 @@ class ScaffoldState extends State<Scaffold> {
...
-      );
-    } else {
-      application = new CustomMultiChildLayout(
-        children: children,
-        delegate: new _ScaffoldLayout(
-          padding: padding
-        )
-      );
-    }
+    EdgeInsets appPadding = (config.appBarBehavior != AppBarBehavior.anchor) ?
```
Comment (dragostis) [inline packages/flutter/lib/src/material/scaffold.dart]: This line could be in the `if` above.

**Comment by dragostis** [comment]
You should probably reword the commit message. LGTM.

File: packages/flutter/lib/src/material/scaffold.dart [hunk 3]
```diff
@@ -591,6 +591,7 @@ class ScaffoldState extends State<Scaffold> {
 
   bool _handleScrollNotification(ScrollNotification notification) {
     final ScrollableState scrollable = notification.scrollable;
+
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/scaffold.dart]: Did you mean to add a blank line here?

**Comment by HansMuller** [comment]
LGTM with a test.

**Comment by jason-simmons** [comment]
Added a test

**Comment by HansMuller** [comment]
It would be useful to put a pointer to the issue in the test, since it's a regression test.

LGTM

**Pull request merged**